examples/collection-get-item/index.tsx

```diff
@@ -0,0 +1,26 @@
+import "./style.css";
+import * as Ariakit from "@ariakit/react";
+
+type CustomItem = {
+  id: string;
+  element?: HTMLElement | null | undefined;
+  color?: string;
+};
+
+export default function Example() {
+  const collection = Ariakit.useCollectionStore<CustomItem>({ items: [] });
+  const renderedItems = collection.useState((state) => state.renderedItems);
+  const purpleItems = renderedItems.filter((item) => item.color === "purple");
+  return (
+    <Ariakit.Collection store={collection} className="collection">
+      <div>Purple items: {purpleItems.length}</div>
+      <Ariakit.CollectionItem>🍎 Apple</Ariakit.CollectionItem>
+      <Ariakit.CollectionItem
+        getItem={(item) => ({ ...item, color: "purple" })}
```

Comment:
[`getItem`](https://ariakit.org/reference/collection-item#getitem) must be a memoized callback, probably outside of the return statement.

Reply:
Oh, fine. I will fix that.

Just for curious, there's a reason for that? If it's only for optimization it seems it will take more work to memoize than that raw function.

Reply:
`getItem` will trigger the item registration each time it changes, which in turn updates the `items` and `renderedItems` states.

This is a very low-level API. In most instances, it should be abstracted into a custom item component. Other components such as `CompositeItem`, `ComboboxItem`, and so on, use it under the hood.

Reply:
Oh got it. I will fix later when on pc